devdoc/worker_thread_requirements.md

```diff
@@ -5,11 +5,19 @@
 `worker_thread` is a helper module that implements execution of a work item on a single separate thread. It can be used in batching operations, sequencing completions, etc.
 The user instantiates a worker thread instance and then signals when the work shall be executed by calling `worker_thread_schedule_process`.
 
+## Thread Concerns
+
+A calls to `worker_thread_schedule_process` must wait for the resulting `WORKER_FUNC` call or the call will not be delivered.  For this reason `worker_thread_schedule_process` function should be called in a single threaded manner.
```

Comment:
>should be called in a single threaded manner [](http://example.com/codeflow?start=185&length=44)

I believe it should enforce that.

What would happen if it would collect the threadId at the time of the `worker_thread_schedule_process` and then compare it (if `WORKER_FUNC` is still executing) with the thread id of the second call to `worker_thread_schedule_process`?

Reply:
It only creates 1 thread (in open), so the thread Id would be the same.